docs/learn/tutorials/versioned/samza-event-hubs-standalone.md

```diff
@@ -0,0 +1,142 @@
+
+#### Get the Code
+
+Let's get started by cloning the hello-samza project
+
+{% highlight bash %}
+git clone https://git.apache.org/samza-hello-samza.git hello-samza
+cd hello-samza
+git checkout latest
+{% endhighlight %}
+
+The project comes up with numerous examples and for this tutorial, we will pick the Azure application.
+
+#### Setting up the Deployment Environment
+
+For our Azure application, we require [ZooKeeper](http://zookeeper.apache.org/). The hello-samza project comes with a script called "grid" to help with the environment setup
+
+{% highlight bash %}
+./bin/grid standalone
+{% endhighlight %}
+
+This command will download, install, and start ZooKeeper and Kafka. It will also check out the latest version of Samza and build it. All package files will be put in a sub-directory called "deploy" inside hello-samza's root folder.
+
+If you get a complaint that JAVA_HOME is not set, then you'll need to set it to the path where Java is installed on your system.
+
+
+#### Configuring the Azure application
+
+Here are the [Event Hubs configs](../../documentation/versioned/connectors/eventhubs.html) you must set before building the project. 
+Configure these in the `src/main/java/samza/examples/AzureApplication.java` file.
+
+{% highlight java %}
+@Override
+public void describe(StreamApplicationDescriptor appDescriptor) {
+  // Define your system here
+  EventHubsSystemDescriptor systemDescriptor = new EventHubsSystemDescriptor("eventhubs");
+
+  // Choose your serializer/deserializer for the EventData payload
+  StringSerde serde = new StringSerde();
+
+  // Define the input and output descriptors with respective configs
+  EventHubsInputDescriptor<KV<String, String>> inputDescriptor =
+      systemDescriptor.getInputDescriptor(INPUT_STREAM_ID, EVENTHUBS_NAMESPACE, EVENTHUBS_INPUT_ENTITY, serde)
+          .withSasKeyName(EVENTHUBS_SAS_KEY_NAME)
+          .withSasKey(EVENTHUBS_SAS_KEY_TOKEN);
+
+  EventHubsOutputDescriptor<KV<String, String>> outputDescriptor =
+      systemDescriptor.getOutputDescriptor(OUTPUT_STREAM_ID, EVENTHUBS_NAMESPACE, EVENTHUBS_OUTPUT_ENTITY, serde)
+          .withSasKeyName(EVENTHUBS_SAS_KEY_NAME)
+          .withSasKey(EVENTHUBS_SAS_KEY_TOKEN);
+
+  // Define the input and output streams with descriptors
+  MessageStream<KV<String, String>> eventhubInput = appDescriptor.getInputStream(inputDescriptor);
+  OutputStream<KV<String, String>> eventhubOutput = appDescriptor.getOutputStream(outputDescriptor);
+
+  //...
+}
+{% endhighlight %}
+
+Alternatively, you man set these properties in in the `src/main/config/azure-application-local-runner.properties` file.
+Note: the keys set in the `.properties` file will override the ones set in code with Descriptors.
+{% highlight jproperties %}
+# define an event hub system factory with your identifier. eg: eh-system
```

Comment:
i'd bias towards removing examples of configuration entirely and keep the description simple